pkg/scheduler/framework/preemption/preemption_test.go

```diff
@@ -404,34 +428,51 @@ func TestPrepareCandidate(t *testing.T) {
 						Containers([]v1.Container{st.MakeContainer().Name("container1").Obj()}).
 						Obj()
 
+		failVictim1WithMatchingCondition = st.MakePod().Name("fail-victim").UID("victim1").
+							Node(node1Name).SchedulerName(defaultSchedulerName).Priority(midPriority).
+							Conditions([]v1.PodCondition{condition}).
+							Containers([]v1.Container{st.MakeContainer().Name("container1").Obj()}).
+							Obj()
+
 		preemptor = st.MakePod().Name("preemptor").UID("preemptor").
 				SchedulerName(defaultSchedulerName).Priority(highPriority).
 				Containers([]v1.Container{st.MakeContainer().Name("container1").Obj()}).
 				Obj()
 	)
 
 	tests := []struct {
-		name           string
-		nodeNames      []string
-		candidate      *fakeCandidate
-		preemptor      *v1.Pod
-		testPods       []*v1.Pod
+		name                  string
+		nodeNames             []string
+		candidate             *fakeCandidate
+		preemptor             *v1.Pod
+		testPods              []*v1.Pod
+		expectedDeletedPods   []string
+		expectedDeletionError bool
+		expectedPatchError    bool
+		// Only compared when async preemption is disabled.
 		expectedStatus *framework.Status
+		// Only compared when async preemption is enabled.
+		expectedPreemptingMap sets.Set[types.UID]
+		expectedActivatedPods map[string]*v1.Pod
```

Comment:
Do we need a field for this if it's always equal to map[string]*v1.Pod{preemptor.Name: preemptor}? Won't it be possible to to do this check implicitly?